tests/activate.bats

```diff
@@ -81,6 +81,9 @@ setup_file() {
 # ---------------------------------------------------------------------------- #
 
 @test "'flox activate' with eval works with 'dash'" {
+
+  skip "This test fails with exit code 2, we should go after this"
```

Comment:
Are you thinking this test is flaky or due to the PR? Seems like we should figure out why it's failing if it's due to the PR

Reply:
It started with this PR but it' only dash and i didn't figure out why yet.
I assume something related to the activation script but I'm not sure.

Reply:
My guess is that bash passthru
https://github.com/flox/flox/blob/2cff0a2214ad97bd7527733b2cd4729137322420/crates/flox/src/main.rs#L36
is called before `run()`
https://github.com/flox/flox/blob/2cff0a2214ad97bd7527733b2cd4729137322420/crates/flox/src/main.rs#L108
which calls `set_parent_process_id()`. Detecting the shell depends on that variable being set, so that's probably why it's failing

Reply:
isnt flox-bash setting those too?

Reply:
Not that I see
```
> rg FLOX_PARENT_PID
flox-bash/lib/utils.sh
1425:#    in the form of the FLOX_PARENT_PID environment variable
1439:	if [ -n "${FLOX_PARENT_PID:-}" ]; then
1441:		if [ -r "/proc/$FLOX_PARENT_PID/cmdline" ]; then
1443:			# argv[0] rather than /proc/$FLOX_PARENT_PID/exe
1445:			mapfile -d '' cmdline < "/proc/$FLOX_PARENT_PID/cmdline"
1447:		elif local psOutput="$($_ps -c -o command= -p $FLOX_PARENT_PID 2>/dev/null)"; then

crates/flox/src/main.rs
204:/// $FLOX_PARENT_PID is used when launching sub-shells to ensure users
208:    env::set_var("FLOX_PARENT_PID", ppid.to_string());
```

Reply:
dash seems to work now, thanks for the pointer

Reply:
the develop test also works which might or might not be related 🤔